perfkitbenchmarker/benchmarks/fio_benchmark.py

```diff
@@ -53,6 +57,95 @@
                   'scratch_disk': True,
                   'num_machines': 1}
 
+GLOBALS_TEMPLATE = """
+[global]
+ioengine=libaio
+invalidate=1
+blocksize=4k
+direct=1
+runtime=10m
+time_based
+filename={filename}
+do_verify=0
+verify_fatal=0
+randrepeat=0
+size={size}
+
+"""
+
+SINGLE_JOB_TEMPLATE = """
+[{rwkind}-io-depth-{iodepth}]
+stonewall
+rw={rwkind}
+iodepth={iodepth}
+
+"""
+
+
+def GetIODepths(io_depths):
+  """Parse the io_depths parameter.
+
+  Args:
+    io_depths: a string in the format of the --io_depths flag.
+
+  Returns:
+    An iterable of integers.
+
+  Raises:
+    ValueError if FLAGS.io_depths doesn't follow a format it recognizes.
+  """
+
+  try:
+    return [int(io_depths)]
+  except ValueError:
+    bounds = io_depths.split('-', 1)
+
+    if len(bounds) != 2:
+      raise ValueError
+
+    return range(int(bounds[0]), int(bounds[1]) + 1)
+
+
+def WriteJobFile(mount_point):
+  """Write a fio job file.
+
+  Args:
+    mount_point: the mount point of the disk we're testing against.
+
+  Returns:
+    The contents of a fio job file, as a string.
+  """
+
+  if FLAGS.against_device:
+    filename = mount_point
+    size = FLAGS.device_fill_size
+  else:
+    filename = os.path.join(mount_point, 'fio-temp-file')
```

Comment:
Also, please use `posixpath.join` to create remote file paths to maintain support for a windows controller.